core/kernel/ldelf_loader.c

```diff
@@ -408,28 +426,27 @@ TEE_Result ldelf_dlsym(struct user_mode_ctx *uctx, TEE_UUID *uuid,
 	uint32_t panicked = 0;
 	size_t len = strnlen(sym, maxlen);
 	struct ts_session *sess = NULL;
+	struct dl_entry_arg *dl_entry_arg = NULL;
+	size_t len_arg_with_sym = sizeof(*arg) + len + 1;
 
 	if (len == maxlen)
 		return TEE_ERROR_BAD_PARAMETERS;
 
 	usr_stack -= ROUNDUP(sizeof(*arg) + len + 1, STACK_ALIGNMENT);
 	arg = (struct dl_entry_arg *)usr_stack;
 
-	res = vm_check_access_rights(uctx,
-				     TEE_MEMORY_ACCESS_READ |
-				     TEE_MEMORY_ACCESS_WRITE |
-				     TEE_MEMORY_ACCESS_ANY_OWNER,
-				     (uaddr_t)arg, sizeof(*arg) + len + 1);
-	if (res) {
-		EMSG("ldelf stack is inaccessible!");
-		return res;
-	}
+	dl_entry_arg = malloc(len_arg_with_sym);
+	if (!dl_entry_arg)
+		return TEE_ERROR_OUT_OF_MEMORY;
 
-	memset(arg, 0, sizeof(*arg));
-	arg->cmd = LDELF_DL_ENTRY_DLSYM;
-	arg->dlsym.uuid = *uuid;
-	memcpy(arg->dlsym.symbol, sym, len);
-	arg->dlsym.symbol[len] = '\0';
+	dl_entry_arg->cmd = LDELF_DL_ENTRY_DLSYM;
+	dl_entry_arg->dlsym.uuid = *uuid;
+	memcpy(dl_entry_arg->dlsym.symbol, sym, len);
+	dl_entry_arg->dlsym.symbol[len] = '\0';
+
+	res = copy_to_user(arg, dl_entry_arg, len_arg_with_sym);
+	if (res)
```

Comment:
Can remove the `out` branch label by freeing `dl_entry_arg()` right after `copy_to_user()` returns.